rerun_cpp/src/rerun/archetypes/segmentation_image_ext.cpp

```diff
@@ -1,75 +1,56 @@
-#include "segmentation_image.hpp"
-
-#include "../collection_adapter_builtins.hpp"
 #include "../error.hpp"
-
-#include <sstream>
+#include "segmentation_image.hpp"
 
 namespace rerun::archetypes {
 
-#if 0
+#ifdef EDIT_EXTENSION
     // <CODEGEN_COPY_TO_HEADER>
 
-    /// New segmentation image from height/width and tensor buffer.
+#include "../image_utils.hpp"
+
+    /// Row-major. Borrows.
     ///
-    /// \param shape
-    /// Shape of the image. Calls `Error::handle()` if the tensor is not 2-dimensional
-    /// Sets the dimension names to "height" and "width" if they are not specified.
-    /// \param buffer
-    /// The tensor buffer containing the segmentation image data.
-    SegmentationImage(Collection<datatypes::TensorDimension> shape, datatypes::TensorBuffer buffer)
-        : SegmentationImage(datatypes::TensorData(std::move(shape), std::move(buffer))) {}
+    /// The length of the data should be `W * H`.
+    template <typename TElement>
+    SegmentationImage(const TElement* pixels, components::Resolution2D resolution_)
+        : SegmentationImage{
+              reinterpret_cast<const uint8_t*>(pixels), resolution_, get_data_type(pixels)} {}
 
-    /// New segmentation image from tensor data.
+    /// Row-major.
```

Comment:
these are pretty sporadic, would be nice to write something nicer imho. Same on DepthImage, but didn't pay enough attention last time.

* what is row major?
* what does "borrows" mean? That's not a common concept in C++ and the implication (crashes at random) needs to be clarified

Reply:
I'll add a TODO to https://github.com/rerun-io/rerun/issues/6386